github_backup/github_backup.py

```diff
@@ -0,0 +1,1143 @@
+        output_file = "{0}/starred.json".format(account_cwd)
+        template = "https://{0}/users/{1}/starred".format(get_github_api_host(args), args.user)
+        _backup_data(args,
+                     "starred repositories",
+                     template,
+                     output_file,
+                     account_cwd)
+
+    if args.include_watched or args.include_everything:
+        output_file = "{0}/watched.json".format(account_cwd)
+        template = "https://{0}/users/{1}/subscriptions".format(get_github_api_host(args), args.user)
+        _backup_data(args,
+                     "watched repositories",
+                     template,
+                     output_file,
+                     account_cwd)
+
+    if args.include_followers or args.include_everything:
+        output_file = "{0}/followers.json".format(account_cwd)
+        template = "https://{0}/users/{1}/followers".format(get_github_api_host(args), args.user)
+        _backup_data(args,
+                     "followers",
+                     template,
+                     output_file,
+                     account_cwd)
+
+    if args.include_following or args.include_everything:
+        output_file = "{0}/following.json".format(account_cwd)
+        template = "https://{0}/users/{1}/following".format(get_github_api_host(args), args.user)
+        _backup_data(args,
+                     "following",
+                     template,
+                     output_file,
+                     account_cwd)
+
+
+def _backup_data(args, name, template, output_file, output_directory):
+    skip_existing = args.skip_existing
+    if not skip_existing or not os.path.exists(output_file):
+        log_info('Retrieving {0} {1}'.format(args.user, name))
+        mkdir_p(output_directory)
+        data = retrieve_data(args, template)
+
+        log_info('Writing {0} {1} to disk'.format(len(data), name))
+        with codecs.open(output_file, 'w', encoding='utf-8') as f:
+            json_dump(data, f)
+
+
+def json_dump(data, output_file):
+    json.dump(data,
+              output_file,
+              ensure_ascii=False,
+              sort_keys=True,
+              indent=4,
+              separators=(',', ': '))
+
+
+def main():
+    args = parse_args()
+
+    output_directory = os.path.realpath(args.output_directory)
+    if not os.path.isdir(output_directory):
+        log_info('Create output directory {0}'.format(output_directory))
+        mkdir_p(output_directory)
+
+    if args.lfs_clone:
+        check_git_lfs_install()
+
+    log_info('Backing up user {0} to {1}'.format(args.user, output_directory))
+
+    authenticated_user = get_authenticated_user(args)
+    repositories = retrieve_repositories(args, authenticated_user)
+    repositories = filter_repositories(args, repositories)
+    backup_repositories(args, output_directory, repositories)
+    backup_account(args, output_directory)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
remove this, the file shouldn't be "executable" in this sense.

Reply:
alright; that makes sense